src/vs/workbench/parts/terminal/electron-browser/terminalPanel.ts

```diff
@@ -167,7 +168,14 @@ export class TerminalPanel extends Panel {
 		if (!themeId) {
 			themeId = this.themeService.getTheme();
 		}
-		let theme = this.configurationHelper.getTheme(themeId);
+
+		let terminalThemeId = this.configurationHelper.getTerminalThemeId(themeId);
```

Comment:
Can you pull `getBaseThemeId` into this file and check it here instead of calling into `configurationHelper` twice?